### PR TITLE
feat: [CDS-35216]: Add support for custom artifact with fixed value

### PR DESCRIPTION
**Pull request by abosii-harness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31673)

### CODE REVIEW

File: 127-cd-nextgen-beans/src/main/java/io/harness/cdng/artifact/outcome/CustomArtifactOutcome.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.springframework.data.annotation.TypeAlias;
+
+@Value
+@Builder
+@EqualsAndHashCode(callSuper = false)
+@TypeAlias("customArtifactOutcome")
+@JsonTypeName("customArtifactOutcome")
+@OwnedBy(CDC)
```
Comment (shubham-in10se) [inline 127-cd-nextgen-beans/src/main/java/io/harness/cdng/artifact/outcome/CustomArtifactOutcome.java:28]: shouldn't this be CDP

File: 125-cd-nextgen/src/main/java/io/harness/cdng/artifact/bean/yaml/CustomArtifactConfig.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.experimental.Wither;
+import org.springframework.data.annotation.TypeAlias;
+
+@Data
+@Builder
+@OwnedBy(CDC)
```
Comment (shubham-in10se) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/artifact/bean/yaml/CustomArtifactConfig.java:35]: CDP. ..?

**Review by shubham-in10se — APPROVED** [review]

**Review by abosii-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31673)* status: 23 file(s) to review. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[125-cd-nextgen/src/main/java/io/harness/cdng/artifact/bean/yaml/CustomArtifactConfig.java, line 35 at r1](https://reviewable.io/reviews/harness/harness-core/31673#-MymS_MfXMVUOPlk_vTl-r1-35:-MzOWxEO8h9N09CyBjB1:b-9fud8d) ([raw file](https://github.com/harness/harness-core/blob/27d5acff68b864c42749ac9ec5d86bbb3b985ae0/125-cd-nextgen/src/main/java/io/harness/cdng/artifact/bean/yaml/CustomArtifactConfig.java#L35)):*
<details><summary><i>Previously, shubham-in10se (Shubham Maheshwari) wrote…</i></summary><blockquote>

CDP. ..?
</blockquote></details>

All artifacts are owned by CDC
___
*[127-cd-nextgen-beans/src/main/java/io/harness/cdng/artifact/outcome/CustomArtifactOutcome.java, line 28 at r1](https://reviewable.io/reviews/harness/harness-core/31673#-MymS_MfXMVUOPlk_vTp-r1-28:-MzOX02lE1DAgv2L5qtL:b-9fud8d) ([raw file](https://github.com/harness/harness-core/blob/27d5acff68b864c42749ac9ec5d86bbb3b985ae0/127-cd-nextgen-beans/src/main/java/io/harness/cdng/artifact/outcome/CustomArtifactOutcome.java#L28)):*
<details><summary><i>Previously, shubham-in10se (Shubham Maheshwari) wrote…</i></summary><blockquote>

shouldn't this be CDP
</blockquote></details>

All artifacts are owned by CDC

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/artifact/PrimaryArtifactPlanCreator.java [hunk 3]
```diff
@@ -53,18 +57,28 @@ public PlanCreationResponse createPlanForField(PlanCreationContext ctx, PrimaryA
         ctx.getDependency().getMetadataMap().get(YamlTypes.UUID).toByteArray());
     StepParameters stepParameters = (StepParameters) kryoSerializer.asInflatedObject(
         ctx.getDependency().getMetadataMap().get(PlanCreatorConstants.PRIMARY_STEP_PARAMETERS).toByteArray());
+    ArtifactStepParameters artifactStepParameters = (ArtifactStepParameters) stepParameters;
+
+    StepType stepType;
+    String facilitatorType;
+    if (ArtifactSourceType.CUSTOM_ARTIFACT == artifactStepParameters.getType()) {
```
[the remark this answers is not in the document]
Comment (abosii-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/artifact/PrimaryArtifactPlanCreator.java]: Done

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 14 of 23 files at r1, 1 of 7 files at r2, 3 of 7 files at r3, 2 of 4 files at r4, 5 of 5 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31673)* status: 2. unresolved discussions.

**Review by anilchowdhury-harness — APPROVED** [review]
:lgtm:

> Reviewed 14 of 23 files at r1, 1 of 7 files at r2, 3 of 7 files at r3, 2 of 4 files at r4, 5 of 5 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31673#-:-N-2rxAGADQVImaKu2vl:bnfp4nl)* status: 2. unresolved discussions.

**Review by hextechpal — APPROVED** [review]
> Reviewed 14 of 23 files at r1, 1 of 7 files at r2, 3 of 7 files at r3, 2 of 4 files at r4, 5 of 5 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31673)* status: :shipit: complete! ✓

**Pull request merged**